data-prepper-plugins/s3-source/src/test/java/com/amazon/dataprepper/plugins/source/CompressionEngineTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source;
+
+import com.amazon.dataprepper.plugins.source.configuration.CompressionOption;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class CompressionEngineTest {
+    private CompressionEngine compressionEngine;
+    private CompressionOption compressionOption;
+    private ResponseInputStream<GetObjectResponse> responseInputStream;
+    private String s3Key;
+
+    @BeforeEach
+    void setUp() {
+        s3Key = UUID.randomUUID().toString();
+        responseInputStream = mock(ResponseInputStream.class);
+    }
+
+    @Test
+    void createInputStream_with_none_should_return_instance_of_ResponseInputStream() throws IOException {
+        compressionEngine = new CompressionEngine(CompressionOption.NONE);
+        InputStream inputStream = compressionEngine.createInputStream(s3Key, responseInputStream);
+        assertThat(inputStream, instanceOf(ResponseInputStream.class));
```

Comment:
Making this assertion to be `sameInstance(inputStream)` would be a tighter restriction.

Reply:
Nice! I'm not aware of this method 👍